examples/cfd/external_aerodynamics/domino/src/inference_on_stl.py

```diff
@@ -0,0 +1,1496 @@
+        grid = torch.cat((xv, yv, zv), axis=-1)
+        return grid
+
+    def process_surface_mesh(self, bounding_box=None, bounding_box_surface=None):
+        surface_vertices = self.surface_vertices
+        surface_indices = self.surface_indices
+        surface_areas = self.surface_areas
+        surface_centers = self.surface_centers
+
+        start_time = time.time()
+
+        if bounding_box is None:
+            # Create a bounding box
+            s_max = torch.amax(surface_vertices, 0)
+            s_min = torch.amin(surface_vertices, 0)
+
+            c_max = s_max + (s_max - s_min) / 2
+            c_min = s_min - (s_max - s_min) / 2
+            c_min[2] = s_min[2]
+        else:
+            c_min = bounding_box[0]
+            c_max = bounding_box[1]
+
+        if bounding_box_surface is None:
+            # Create a bounding box
+            s_max = torch.amax(surface_vertices, 0)
+            s_min = torch.amin(surface_vertices, 0)
+
+            surf_max = s_max + (s_max - s_min) / 2
+            surf_min = s_min - (s_max - s_min) / 2
+            surf_min[2] = s_min[2]
+        else:
+            surf_min = bounding_box_surface[0]
+            surf_max = bounding_box_surface[1]
+
+        nx, ny, nz = self.grid_resolution
+
+        grid = self.create_grid_torch(c_max, c_min, self.grid_resolution)
+        grid_reshaped = torch.reshape(grid, (nx * ny * nz, 3))
+
+        # SDF on grid
+        sdf_grid = signed_distance_field(
+            surface_vertices, surface_indices, grid_reshaped, device=self.device
+        )
+        sdf_grid = torch.reshape(sdf_grid, (nx, ny, nz))
+
+        surface_areas = torch.unsqueeze(surface_areas, -1)
+        center_of_mass = torch.sum(surface_centers * surface_areas, 0) / torch.sum(
+            surface_areas
+        )
+
+        s_grid = self.create_grid_torch(surf_max, surf_min, self.grid_resolution)
+        surf_grid_reshaped = torch.reshape(s_grid, (nx * ny * nz, 3))
+
+        surf_sdf_grid = signed_distance_field(
+            surface_vertices, surface_indices, surf_grid_reshaped, device=self.device
+        )
+        surf_sdf_grid = torch.reshape(surf_sdf_grid, (nx, ny, nz))
+
+        # Sample surface_vertices
+        geometry_points = self.geom_points_sample
+        surface_vertices = shuffle_array_torch(
+            surface_vertices, geometry_points, device=self.device
+        )
+        if self.normalize_coordinates:
+            grid = 2.0 * (grid - c_min) / (c_max - c_min) - 1.0
+            s_grid = 2.0 * (s_grid - surf_min) / (surf_max - surf_min) - 1.0
+
+        surface_vertices = torch.unsqueeze(surface_vertices, 0)
+        grid = torch.unsqueeze(grid, 0)
+        s_grid = torch.unsqueeze(s_grid, 0)
+        sdf_grid = torch.unsqueeze(sdf_grid, 0)
+        surf_sdf_grid = torch.unsqueeze(surf_sdf_grid, 0)
+        max_min = [c_min, c_max]
+        surf_max_min = [surf_min, surf_max]
+        center_of_mass = center_of_mass
+
+        return (
```

Comment:
Nit: I would try not return 8-element tuple here. Maybe use dict or named tuple or dataclass.